crates/cow-amm/src/cache.rs

```diff
@@ -78,10 +92,27 @@ impl EventStoring<CowAmmEvent> for Storage {
             let CowAmmEvent::CowammpoolCreated(cow_amm) = event.data;
             let cow_amm = cow_amm.amm;
 
+            let amm = Amm::new(cow_amm, &self.0.helper).await?;
```

Comment:
> I believe the maintenance loop ensures sequential execution of maintenance jobs

Doesn't [join_all](https://github.com/cowprotocol/services/blob/74a1271c344e057a81870569216b803b832f676c/crates/shared/src/maintenance.rs#L40) execute everything concurrently without guaranteeing ordering.

> I'd still say we should remove this (or at least call a function on EmptyPoolRemoval that checks balances for a single pool to not duplicate that part of the business logic here). It might not seem like a lot of logic in the current state, but these are the kinds of things that grow over time and turn components into things that are messy and hard to reason about.

Ok, removing it for now. In case we still have a lot of errors, maybe it would make sense to circle back.

Reply:
> Doesn't [join_all](https://github.com/cowprotocol/services/blob/74a1271c344e057a81870569216b803b832f676c/crates/shared/src/maintenance.rs#L40) execute everything concurrently without guaranteeing ordering.

Ah you are right (I should have checked the code). I guess our idea was that we could right a single maintainer that guarantees sequentiality between its operation.